Tasks/Maven/maven.ps1

```diff
@@ -28,14 +32,37 @@ import-module "Microsoft.TeamFoundation.DistributedTask.Task.Internal"
 import-module "Microsoft.TeamFoundation.DistributedTask.Task.Common"
 import-module "Microsoft.TeamFoundation.DistributedTask.Task.TestResults"
 
-if($jdkVersion -and $jdkVersion -ne "default")
+# If JAVA_HOME is being set by choosing a JDK version find the path to that specified version else use the path given by the user
+$jdkPath = $null
+if($javaHomeSelection -eq 'JDKVersion')
 {
-    $jdkPath = Get-JavaDevelopmentKitPath -Version $jdkVersion -Arch $jdkArchitecture
-    if (!$jdkPath) 
+    Write-Verbose "Using JDK version to find and set JAVA_HOME"
+    # If the JDK version is not the deafult set the jdkPath to the new JDK version selected
+    if($jdkVersion -and ($jdkVersion -ne "default"))
     {
-        throw "Could not find JDK $jdkVersion $jdkArchitecture, please make sure the selected JDK is installed properly"
+        $jdkPath = Get-JavaDevelopmentKitPath -Version $jdkVersion -Arch $jdkArchitecture
+        if (!$jdkPath) 
+        {
+            throw (Get-LocalizedString -Key 'Could not find JDK {0} {1}. Please make sure the selected JDK is installed properly.' -ArgumentList $jdkVersion, $jdkArchitecture)
+        }
     }
+}
+else
+{
+    Write-Verbose "Using path from user input to set JAVA_HOME"
+    if($jdkUserInputPath -and (Test-Path -Path $jdkUserInputPath))
```

Comment:
Here's a really subtle bug, albeit probably not very likely to occur. Unless your path has wildcards in it, you should use -LiteralPath. Also if you wanted to you could use -Type Container or -Type Leaf... probably not going to make a difference.

The problem with powershell is that it supports more wildcards than DOS. DOS only supports * and ? which are both not valid file name characters. PowerShell additionally supports character ranges using "[" and "]" syntax, which are valid file name characters. For example, create a file c:\foo[1].txt and Test-Path -Path 'c:\foo[1].txt' will tell you the file doesn't exist.